### DISCUSSION TITLE
What is 'error = failed to parse http request: Version'?

### DISCUSSION THREAD

**Discussion by kination** [tokio-rs/tokio#3917] [Q&A]
Hello,
I'm making some demo application which receives http request, using `tinyhttp.rs` example.
But I'm receiving message something as:
```
request: Ok(Request { method: POST, uri: /, version: HTTP/1.1, headers: {"host": "127.0.0.1:7700", "user-agent": "curl/7.64.1", "accept": "*/*", "content-type": "application/json", "content-length": "18"}, body: () })
request: Err(Custom { kind: Other, error: "failed to parse http request: Version" })
```

I've tested with:
```
$ curl -X POST -H "Content-Type: application/json" -d '{ "data": "test" }' 127.0.0.1:7700
```

Is there some reason?

**Comment by Darksonn** [comment]
I believe this is because the `tinyhttp.rs` example doesn't properly handle request bodies, so it starts reading the next request on the connection at the wrong place in the request.
